### PR TITLE
fix(ingest/mode): add connection timeouts to avoid RemoteDisconnected errors

### PR DESCRIPTION
**Pull request by sagar-salvi-apptware**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/mode.py [hunk 1]
```diff
@@ -1469,16 +1483,21 @@ def _get_request_json(self, url: str) -> Dict:
...
-                response = self.session.get(url)
-                response.raise_for_status()
+                response = self.session.get(
+                    url, timeout=self.config.api_options.timeout
+                )
                 return response.json()
+            except ConnectionError as e:
+                # TODO Need to check why server is closes the connection before sending a complete response
+                logging.warning(f"RemoteDisconnected or ProtocolError: {str(e)}")
+                return {}
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/mode.py:1500]: if we don't get any info from the API, does it really make sense to pretend like the API returned nothing?

File: metadata-ingestion/tests/integration/mode/test_mode.py [hunk 2]
```diff
@@ -41,12 +41,17 @@ def __init__(self, error_list, status_code):
...
     def json(self):
         return self.json_data
 
-    def get(self, url):
+    def mount(self, prefix, adaptor):
+        return self
+
+    def get(self, url, timeout=40):
         self.url = url
+        self.timeout = timeout
```
Comment (hsheth2) [inline metadata-ingestion/tests/integration/mode/test_mode.py:54]: what's going on here?

File: metadata-ingestion/src/datahub/ingestion/source/mode.py [hunk 3]
```diff
@@ -1469,16 +1483,21 @@ def _get_request_json(self, url: str) -> Dict:
...
         @r.wraps
         def get_request():
             try:
-                response = self.session.get(url)
-                response.raise_for_status()
+                response = self.session.get(
+                    url, timeout=self.config.api_options.timeout
+                )
                 return response.json()
+            except ConnectionError as e:
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/mode.py:1497]: also if we have `retry_if_exception_type` for ConnectionError, then we should not be suppressing the ConnectionError exception

**Review by hsheth2 — CHANGES_REQUESTED** [review]

File: metadata-ingestion/tests/integration/mode/test_mode.py (hunk 2 above)
Comment (sagar-salvi-apptware) [reply metadata-ingestion/tests/integration/mode/test_mode.py:54]: not required removed !!!

File: metadata-ingestion/src/datahub/ingestion/source/mode.py (hunk 3 above)
Comment (sagar-salvi-apptware) [reply metadata-ingestion/src/datahub/ingestion/source/mode.py:1497]: For debug purpose i  have added this but not required removing this as we already added `retry_if_exception_type`

File: metadata-ingestion/src/datahub/ingestion/source/mode.py (hunk 1 above)
Comment (sagar-salvi-apptware) [reply metadata-ingestion/src/datahub/ingestion/source/mode.py]: Idea is to debug, dont need to pretend here. Removed this !!!

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**